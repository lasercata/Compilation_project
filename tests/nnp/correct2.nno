//----------------------------------------------------
// Programme correct
//   
//
//    Affiche 3 puis 4
//    Puis get x et print x+1
//
//------------------------------------------------------
procedure pp is
  procedure affiche(x,y : integer) is
    i,j:integer;
  begin
    put(x);
    put(y)
  end;
  a,b,c : integer;
begin
  affiche(3,4);
  get(a);
  b:=a+1;
  put(b)
end.
