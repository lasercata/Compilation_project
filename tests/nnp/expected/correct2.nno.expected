debutProg()
tra(11)
reserver(2)
empilerAd(0)
valeurPile()
put()
empilerAd(1)
valeurPile()
put()
retourProc()
reserver(3)
reserverBloc()
empiler(3)
empiler(4)
traStat(3,2)
empiler(0)
get()
empiler(1)
empiler(0)
valeurPile()
empiler(1)
add()
affectation()
empiler(1)
valeurPile()
put()
finProg()